scripts/deep_analysis_example.ipynb

```diff
@@ -0,0 +1,251 @@
+    "    num_loops = len(json)\n",
+    "    for loop_results in json:\n",
+    "        for frame_index, frame_results in enumerate(loop_results[\"per_frame_results\"]):\n",
+    "            if frame_index >= len(results_per_frame):\n",
+    "                results_per_frame.append(defaultdict(int))\n",
+    "            results_per_frame[frame_index]['sequence_time_ns'] = frame_results['sequence_time_ns']\n",
+    "            for command_buffer_timings in frame_results[\n",
+    "                \"command_buffer_timings\"\n",
+    "            ].values():\n",
+    "                for scope_name, scope_timings in command_buffer_timings[\n",
+    "                    \"scope_timings\"\n",
+    "                ].items():\n",
+    "                    for scope_timing in scope_timings:\n",
+    "                        results_per_frame[frame_index][scope_name] += (\n",
+    "                            scope_timing[\"end\"] - scope_timing[\"start\"]\n",
+    "                        ) / num_loops / 1_000_000 # in ms\n",
+    "            for metric_name, metric in frame_results[\"metrics\"].items():\n",
+    "                # TODO: Flatten this in rust to fan_speed_rpm\n",
+    "                if metric_name == \"fan_speed\":\n",
+    "                    value = metric[\"Percent\"] if \"Percent\" in metric else metric[\"Rpm\"]\n",
+    "                    results_per_frame[frame_index][\"fan_speed_rpm\"] += (\n",
+    "                        value / num_loops\n",
+    "                    )\n",
+    "                # Filter out unavailable data and the timestamp\n",
+    "                elif metric is not None and metric_name != \"timestamp\":\n",
+    "                    results_per_frame[frame_index][metric_name] += metric / num_loops\n",
+    "    # TODO: Aggregate CPU timings\n",
+    "    return pd.DataFrame([flatten(x) for x in results_per_frame])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "939b04cb-7ffd-4963-8dac-71c0a76275cf",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "files = [\n",
+    "    '../../a_ray_tracing_inline.csv',\n",
+    "    '../../a_ray_tracing_pipeline.csv',\n",
+    "    '../../b_ray_tracing_inline.csv',\n",
+    "    '../../b_ray_tracing_pipeline.csv',\n",
+    "]\n",
+    "\n",
+    "scores = {}\n",
+    "\n",
+    "for file in files:\n",
+    "    # Read csv\n",
+    "    file_scores = pd.read_csv(file)\n",
+    "    # Remove empty results\n",
+    "    file_scores = file_scores.loc[:, (file_scores != 0).any(axis=0)]\n",
+    "    scores[file] = file_scores\n",
+    "scores = pd.concat(scores).groupby(level=0, sort=False).mean().T.drop('Loop', errors=\"ignore\")\n",
+    "scores"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2fbaaf11-5b32-43ed-84d1-8894ec6de013",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "scores.plot(kind=\"bar\", figsize=(20,5), colormap='Dark2', grid=True, rot=0)"
```

Comment:
Could we add plotting of all the different score categories. So compare the following: 

- `a_raytracing_inline`, `a_raytracing_pipeline` vs `b_raytracing_inline` vs `b_raytracing_pipeline` 
- `a_rasterization_inline`, `a_rasterization_pipeline` vs `b_rasterization_inline` vs `b_rasterization_pipeline` 
- etc..

Reply:
Added it as a separate notebook.